### PR TITLE
NONAME-201 add buttons for pagination

### PR DESCRIPTION
**Pull request by K4RP4T**

### CODE REVIEW

**Review by yanvoi — APPROVED** [review]

File: backend/src/api/models/user_groups/user_groups.py [hunk 1]
```diff
@@ -66,21 +75,25 @@ def validate_user_public_key(cls, v, info: ValidationInfo) -> str:
...
 
-    class Config:
-        extra = "forbid"
 
+class UserGroupUsersActions(UserGroupAction):
+    users: list[str]
 
-class AllUserGroupsOutput(BaseModel):
+    @field_validator("users")
+    def validate_user_public_key(cls, v, info: ValidationInfo) -> str:
```
Comment (mtszlachetka) [inline backend/src/api/models/user_groups/user_groups.py:83]: shouldn't this be called validate_emails?

File: backend/src/api/models/user_groups/user_groups.py [hunk 2]
```diff
@@ -55,6 +51,19 @@ def validate_user_public_key(cls, v, info: ValidationInfo) -> str:
         return v
 
 
+class UserGroupMultipleActions(UserGroupCreator):
+    names: list[str]
+
+    @field_validator("names")
+    def validate_user_public_key(cls, v, info: ValidationInfo) -> str:
```
Comment (mtszlachetka) [inline backend/src/api/models/user_groups/user_groups.py:58]: here too

File: backend/src/api/routes/survey_drafts.py [hunk 3]
```diff
@@ -84,28 +85,24 @@ async def get_survey_draft(
...
-        raise HTTPException(status_code=404, detail="Survey Draft does not exist")
 
-    if survey_draft.creator_id != user.id:
+    if len(deleted_survey_drafts) != len(survey_draft_delete.ids):
         raise HTTPException(
-            status_code=403,
-            detail="User does not have access to this Survey Draft",
+            status_code=404,
+            detail="Some of the survey drafts do not exist or are already deleted",
         )
```
Comment (mtszlachetka) [inline backend/src/api/routes/survey_drafts.py:104]: Is this error needed? The data gets deleted anyway, so the nonexistent or already deleted stuff can just be ignored.

File: backend/src/db/crud/survey.py [hunk 4]
```diff
@@ -20,19 +20,31 @@ def get_survey_by_code(survey_code: str, session: Session) -> Survey:
...
+def delete_surveys(
+    user_id: int, survey_codes: list[str], session: Session
+) -> list[Survey]:
+    statement = select(Survey).where(
+        (Survey.creator_id == user_id)
+        & (Survey.survey_code.in_(survey_codes))
+        & (Survey.is_deleted == False)  # noqa: E712
+    )
+    surveys = session.exec(statement).all()
+    print(surveys)
```
Comment (mtszlachetka) [inline backend/src/db/crud/survey.py:32]: leftover print?

Comment (yanvoi) [reply backend/src/db/crud/survey.py]: Deleted

File: backend/src/api/models/user_groups/user_groups.py (hunk 2 above)
Comment (yanvoi) [reply backend/src/api/models/user_groups/user_groups.py]: Done

File: backend/src/api/models/user_groups/user_groups.py (hunk 1 above)
Comment (yanvoi) [reply backend/src/api/models/user_groups/user_groups.py]: Done

File: frontend/src/lib/server/database.ts [hunk 5]
```diff
@@ -105,10 +135,10 @@ export const getSurveyRespondents = (survey_code: string, page: number) => {
 	});
 };
 
-export const deleteSurvey = (user_email: string, survey_code: string) => {
+export const deleteSurveys = (user_email: string, survey_codes: string) => {
```
Comment (mtszlachetka) [inline frontend/src/lib/server/database.ts:138]: shouldn't survey_codes be an array of strings?

File: frontend/src/lib/server/database.ts [hunk 6]
```diff
@@ -142,11 +172,21 @@ export const giveAccessToSurvey = (
 export const takeAwayAccessToSurvey = (
 	user_email: string,
 	survey_code: string,
-	user_email_to_take_access_from: string
+	user_emails: string
```
Comment (mtszlachetka) [inline frontend/src/lib/server/database.ts:175]: same here

File: frontend/src/lib/server/database.ts [hunk 7]
```diff
@@ -219,10 +279,10 @@ export const renameUserGroup = (user_email: string, name: string, new_name: stri
 	});
 };
 
-export const deleteUserGroup = (user_email: string, name: string) => {
+export const deleteUserGroups = (user_email: string, names: string) => {
```
Comment (mtszlachetka) [inline frontend/src/lib/server/database.ts:282]: and here

**Review by mtszlachetka — APPROVED** [review]

**Pull request merged**